storage-manager/src/Ownership.cpp

```diff
@@ -124,28 +138,24 @@ bf::path Ownership::get(const bf::path& p, bool getOwnership)
   return ret;
 }
 
-// minor timesaver
-#define TOUCH(p, f)                                                                                       \
-  {                                                                                                       \
-    int fd = ::open((metadataPrefix / p / f).string().c_str(), O_TRUNC | O_CREAT | O_WRONLY, 0660);       \
-    if (fd >= 0)                                                                                          \
-      ::close(fd);                                                                                        \
-    else                                                                                                  \
-    {                                                                                                     \
-      char buf[80];                                                                                       \
-      int saved_errno = errno;                                                                            \
-      cerr << "failed to touch " << metadataPrefix / p / f << " got " << strerror_r(saved_errno, buf, 80) \
-           << endl;                                                                                       \
-    }                                                                                                     \
-  }
-
-#define DELETE(p, f) ::unlink((metadataPrefix / p / f).string().c_str());
-
 void Ownership::touchFlushing(const bf::path& prefix, volatile bool* doneFlushing) const
 {
   while (!*doneFlushing)
   {
-    TOUCH(prefix, "FLUSHING");
+    {
+      auto kvStorage = KVStorageInitializer::getStorageInstance();
+      auto tnx = kvStorage->createTransaction();
+      const std::string flushingKey = KVPrefixes[static_cast<uint32_t>(KVPrefixId::SM_OWNERSHIP)] +
+                                      prefix.string() +
+                                      ownerShipStates[static_cast<uint32_t>(OwnershipStateId::FLUSHING)];
+      tnx->set(flushingKey, "");
+      if (!tnx->commit())
+      {
+        const char* msg = "Ownership: commit `touchFlushing` failed ";
+        logger->log(LOG_CRIT, msg);
+        throw runtime_error(msg);
```

Comment:
I see that previously DELETE doesn't throw and now it throws. Did you test this control path? SM might get SIGABRT that it doesn't handle that effectively crashes SM.

Reply:
I see only one commit operation  which is not handled at line 304, good catch, thanks.
BTW - each transaction could have some `get/set/remove` operations in a row, and should end with `commit` operation. Only `commit` could fail, but not `get/set/remove` according to C API FDB.
Handling exception is responsibility of Ownership user. It is used in `IOCoordinator` which implements a `read/write/open` operation for the given filename. IOC itself used in Tasks - ReadTask, OpenTask and so on.
Each call of that functions a covered by try catch block:
```
   try
    {
      err = ioc->read(cmd->filename, &resp->payload[resp->returnCode], cmd->offset + resp->returnCode,
                      cmd->count - resp->returnCode);
    }
    catch (exception& e)
    {
      logger->log(LOG_ERR, "ReadTask: caught '%s'", e.what());
      errno = EIO;
      err = -1;
    }
```